cmd/eudico/mir.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain"
+	"github.com/filecoin-project/lotus/chain/beacon"
+	"github.com/filecoin-project/lotus/chain/consensus"
+	"github.com/filecoin-project/lotus/chain/consensus/common"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical/actors/sca"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical/actors/subnet"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical/subnet/resolver"
+	"github.com/filecoin-project/lotus/chain/consensus/mir"
+	"github.com/filecoin-project/lotus/chain/gen/genesis"
+	"github.com/filecoin-project/lotus/chain/gen/slashfilter"
+	"github.com/filecoin-project/lotus/chain/stmgr"
+	"github.com/filecoin-project/lotus/chain/store"
+	lcli "github.com/filecoin-project/lotus/cli"
+	cliutil "github.com/filecoin-project/lotus/cli/util"
+	"github.com/filecoin-project/lotus/extern/sector-storage/ffiwrapper"
+	"github.com/filecoin-project/lotus/node"
+	"github.com/filecoin-project/lotus/node/modules/dtypes"
+)
+
+func NewRootMirConsensus(ctx context.Context, sm *stmgr.StateManager, beacon beacon.Schedule, r *resolver.Resolver,
+	verifier ffiwrapper.Verifier, genesis chain.Genesis, netName dtypes.NetworkName) (consensus.Consensus, error) {
+	return mir.NewConsensus(ctx, sm, nil, beacon, r, verifier, genesis, netName)
+}
+
+var mirbftCmd = &cli.Command{
+	Name:  "mir",
+	Usage: "Mir consensus",
+	Subcommands: []*cli.Command{
+		mirbftGenesisCmd,
+		mirbftMinerCmd,
+
+		daemonCmd(node.Options(
+			node.Override(new(consensus.Consensus), NewRootMirConsensus),
+			node.Override(new(store.WeightFunc), mir.Weight),
+			node.Unset(new(*slashfilter.SlashFilter)),
```

Comment:
out of a curiosity, why are we unsetting the slash filter here? (I agree that is not needed, but we haven't unset it in any other consensus) :)

Reply:
It was unset in PoW, that was the reference impl for all next consensus. Should I set SlashFilter in all of them?

Reply:
And in DelegCns there is no SlashFilter at all

Reply:
Not critical, it was just out of a curiosity.